### PR TITLE
DOC 2126: Writing a CorDapp using a template tutorials rewrite

### PR DESCRIPTION
**Pull request by AnnaK-R3**

### CODE REVIEW

**Review by benpesterr3 — COMMENTED** [review]
congratulations - this is a huge huge improvement and is looking fantastic. 

Most of my notes are concerned with: 
* picky tense-related things and some line by line suggestions. 
* I really would prefer to give more of a story about the IOU. I realise the actual tutorial content is about creating IOUs and that is absolutely fine. However, I would like to see a narrative story of this IOU told as a story about a Loan being agreed, and how the most important elements of this loan can be represented in an absolutely minimal Hello World tutorial. So, because a Loan has been agreed, an IOU must be issued. The Amount is 99units of currency. (make it clear that in the most basic hello world, there is no concept of currency yet!) etc. 

I think the idea of a loan agreement is much closer to an actual business logic than this almost abstractly generated IOU. 

So it could be like this: 

PartyA represents a company who have agreed to lend PartyB an amount of money. In order to maintain the obligations of the loan, you are going to use Corda to represent the loan agreement as an IOU on a network. 

Once you have completed this process, you will see how CorDapps can be constructed according to any business logic you choose to apply. This could be an agreement to repay a loan, but it could be anything else that requires binding agreement between two or more parties. 

Or something - probably not exactly that. But a sense of what you are doing, why it makes sense to do it, and what lies beyond...

File: content/en/docs/corda-os/4.8/run-your-cordapp.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      relevancyStatus: "RELEVANT"
+      constraintInfo:
+        constraint:
+          key: "aSq9DsNNvGhYxYyqA9wd2eduEAZ5AXWgJTbTEw3G5d2maAq8vtLE4kZHgCs5jcB1N31cx1hpsLeqG2ngSysVHqcXhbNts6SkRWDaV7xNcr6MtcbufGUchxredBb6"
+      totalStatesAvailable: -1
+      stateTypes: "UNCONSUMED"
+      otherResults: []
+      ```
+
+      This is the transaction issuing our `IOUState` onto a ledger.
```
Comment (taylorharvey-r3) [inline content/en/docs/corda-os/4.8/run-your-cordapp.md:196]:
```suggestion
      This is the transaction issuing your `IOUState` onto a ledger.
```

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md [hunk 2]
```diff
@@ -64,7 +61,7 @@ From the above example, you can see that `Contract` expresses its constraints th
 * Returns silently if it accepts the transaction proposal.
 
 {{< note >}}
-As mentioned in the [Writing the state](hello-world-state.md) tutorial, the above is Corda source code and is therefore written in Kotlin.
+As mentioned in the [Write the state](write-the-state.md) tutorial, the above is Corda source code and is therefore written in Kotlin.
```
Comment (taylorharvey-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-contract.md:64]:
```suggestion
As mentioned in the Modify the state](modify-the-state.md) tutorial, the above is Corda source code and is therefore written in Kotlin.
```

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md [hunk 3]
```diff
@@ -93,15 +90,18 @@ You can visualise this transaction as follows:
 
 {{< figure alt="simple tutorial transaction 2" zoom="/en/images/simple-tutorial-transaction-2.png" >}}
 
-## Defining the IOUContract
+## Output
 
-To write a contract that enforces these constraints, you'll need to modify either `TemplateContract.java` or
-`TemplateContract.kt` by defining an `IOUContract`, as shown in the following code example:
+The following code presents how your template should look like after performing all the steps described below:
```
Comment (taylorharvey-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-contract.md:95]:
```suggestion
The following code represents what your template should look like after performing all the steps described below:
```

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md [hunk 4]
```diff
@@ -194,14 +200,26 @@ public class IOUContract implements Contract {
...
-{{< /note >}}
 
-Let’s walk through this code step by step.
+## Define the IOUContract
 
+To write a contract that enforces these constraints, you'll need to modify either `TemplateContract.java` or
+`TemplateContract.kt` by defining an `IOUContract`, as shown in the following code example:
 
-### The Create command
+1. Depending whether you are working with Java or Kotlin template, perform one of the following steps:
```
Comment (taylorharvey-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-contract.md:209]:
```suggestion
1. Depending on whether you are working with the Java or Kotlin template, perform one of the following steps:
```

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md [hunk 5]
```diff
@@ -194,14 +200,26 @@ public class IOUContract implements Contract {
...
+
+   * For Java, open `TemplateContract.java` from `contracts/src/main/java/com/template/contracts/TemplateContract.java`.
+   * For Kotlin, open `TemplateContract.kt` from `contracts/src/main/kotlin/com/template/contracts/TemplateContract.kt`.
+
+2. Rename the `TemplateContract` class to `IOUContract`.
+
+3. Update the `@BelongsToContract` annotation in the state definition ([Write the state](write-the-state.md)) to specify `IOUContract` class.
```
Comment (taylorharvey-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-contract.md:216]:
```suggestion
3. Update the `@BelongsToContract` annotation in the state definition ([Modify the state](modify-the-state.md)) to specify `IOUContract` class.
```

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 6]
```diff
@@ -345,7 +376,7 @@ public class IOUFlowResponder extends FlowLogic<Void> {
 
 {{< /tabs >}}
 
-As with the `IOUFlow`, our `IOUFlowResponder` flow is a `FlowLogic` subclass where we’ve overridden `FlowLogic.call`.
+As with the `IOUFlow`, your `IOUFlowResponder` flow is a `FlowLogic` subclass where we’ve overridden `FlowLogic.call`.
 
 The flow is annotated with `InitiatedBy(IOUFlow.class)`, which means that your node will invoke
 `IOUFlowResponder.call` when it receives a message from a instance of `IOUFlow` running on another node. This message
```
Comment (taylorharvey-r3) [inline content/en/docs/corda-os/4.8/modify-the-flow.md:382]: I can't add a comment in the last line, but maybe at the end of the Progress so far section we should add a link to the next page with the link text "test our CorDapp"? I think most of the other pages do this. I guess it should just be consistent throughout, one way or the other.

File: content/en/docs/corda-os/4.8/tut-two-party-flow.md [hunk 7]
```diff
@@ -13,80 +13,22 @@ tags:
...
+# Update the flow
 
-
-
-# Updating the flow
-
-After you have written the contract and defined any necessary constraints, as described in [Writing the contract](tut-two-party-contract.md), you now need to update your flow to achieve three things:
+After you have written the contract and defined any necessary constraints, as described in the [Write the contract](tut-two-party-contract.md) section, you now need to update your flow to achieve three things:
 
 * Verify that the transaction proposal you build fulfills the `IOUContract` constraints.
```
Comment (taylorharvey-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-flow.md:23]:
```suggestion
* Verify that the transaction proposal you built fulfills the `IOUContract` constraints.
```

File: content/en/docs/corda-os/4.8/tut-two-party-flow.md [hunk 8]
```diff
@@ -13,80 +13,22 @@ tags:
...
-
-```
-{{% /tab %}}
+To do this, modify the flow that you created earlier as part of [Write the flow](write-the-flow.md).
 
-{{< /tabs >}}
+## Outcome
 
-Next, update `IOUFlow.call` to the following:
+The following code shows how the updated part of IOUFlow file should look like after applying all the changes described later in this document:
```
Comment (taylorharvey-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-flow.md:31]:
```suggestion
The following code shows how the updated part of `IOUFlow` file should look after applying all changes described later in this document:
```

**Review by taylorharvey-r3 — COMMENTED** [review]
Overall this is looking so much better. Great job, Anna! I've left some comments throughout, mostly minor stuff. If you have any questions let me know.

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 9]
```diff
@@ -261,7 +286,7 @@ Each `Command` contains a command type plus a list of public keys. For now, you
 the transaction to be valid, the lender is required to sign the transaction.
 
 
-#### Creating a transaction builder
+#### Create a transaction builder
 
 To actually build the proposed transaction, you need a `TransactionBuilder`. This is a mutable transaction class to
```
Comment (michelemarsden) [inline content/en/docs/corda-os/4.8/modify-the-flow.md:291]:
```suggestion
To build the proposed transaction, you need a `TransactionBuilder`. This is a mutable transaction class to
```

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 10]
```diff
@@ -286,7 +311,7 @@ You sign the transaction using `ServiceHub.signInitialTransaction`, which return
 `SignedTransaction` is an object that pairs a transaction with a list of signatures over that transaction.
 
 
-### Finalising the transaction
+### Finalise the transaction
 
 You now have a valid signed transaction. All that’s left to do is to have the notary sign it, have that recorded
 locally, and then send it to all the relevant parties. Once that happens the transaction will become a permanent part of the
```
Comment (michelemarsden) [inline content/en/docs/corda-os/4.8/modify-the-flow.md:317]: I didn't feel like the sentence flowed. Minor amendment suggested. 
```suggestion
You now have a valid signed transaction. All that’s left to do is have the notary sign it, record that
locally, and then send it to all the relevant parties. The transaction then becomes a permanent part of the
```

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md [hunk 11]
```diff
@@ -253,49 +271,51 @@ the following are true:
...
 
-Secondly, you will define any constraints on the transaction. For example, an issuance transaction would require that the transaction can have no inputs and only a single output.
+You must define any constraints on the transaction. For example, an issuance transaction would require that the transaction can have no inputs and only a single output.
 
 In Kotlin, you use Corda’s built-in `requireThat` blockthese to impose these and the subsequent constraints. `requireThat`
 provides a terse way to write the following:
 
-* If the condition on the right-hand side doesn’t evaluate to true…
-* …throw an `IllegalArgumentException` with the message on the left-hand side
+* If the condition on the right-hand side doesn’t evaluate to true, throw an `IllegalArgumentException` with the message on the left-hand side.
```
Comment (michelemarsden) [inline content/en/docs/corda-os/4.8/tut-two-party-contract.md:292]: Does this need a bullet, now there's only one?

File: content/en/docs/corda-os/4.8/tut-two-party-introduction.md [hunk 12]
```diff
@@ -19,7 +19,7 @@ title: Applying contract constraints
 
 # Applying contract constraints
 
-In the previous steps of the Hello, World! tutorial, you built a CorDapp which enabled you to model IOUs on ledger.
+In the previous steps of the [Writing a CorDapp using a template](writing-a-cordapp-using-a-template) tutorial, you built a CorDapp which enabled you to model IOUs on ledger.
```
Comment (michelemarsden) [inline content/en/docs/corda-os/4.8/tut-two-party-introduction.md:22]: Minor note on consistency - Line 29 refers to 'on the blockchain', but here we talk about 'on ledger'.

**Review by michelemarsden — COMMENTED** [review]
Minor suggested amends from me.

File: content/en/docs/corda-os/4.8/run-your-cordapp.md [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+* Kotlin: [https://github.com/corda/corda-tut1-solution-kotlin](https://github.com/corda/corda-tut1-solution-kotlin)
+
+
+## Next steps
+
+There are a number of improvements you could make to this CorDapp:
+
+* You could add unit tests, using the contract-test and flow-test frameworks.
+* You could change `IOUState.value` from an integer to a proper amount of a given currency.
+* You could add an API, to make it easier to interact with the CorDapp.
```
Comment (erinking-r3) [inline content/en/docs/corda-os/4.8/run-your-cordapp.md:236]: I'd suggest moving the 'you could' to the lead-in sentence for efficiency:

```suggestion
You could make several improvements to this CorDapp. You could:

* Add unit tests, using the contract-test and flow-test frameworks.
* Change `IOUState.value` from an integer to a proper amount of a given currency.
* Add an API, to make it easier to interact with the CorDapp.
```

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md (hunk 3 above)
Comment (erinking-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-contract.md:95]: I think this sentence could be made a bit shorter: 
```suggestion
After you follow all the steps, your CorDapp will look like this:
```

File: content/en/docs/corda-os/4.8/tut-two-party-flow.md (hunk 8 above)
Comment (erinking-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-flow.md:31]: Similar to the above suggestion:

```suggestion
After you have followed all the steps, your IOUFlow file will look like this:
```

File: content/en/docs/corda-os/4.8/tut-two-party-introduction.md (hunk 12 above)
Comment (erinking-r3) [inline content/en/docs/corda-os/4.8/tut-two-party-introduction.md:22]: Tiny, but there's an opportunity for simpler language here:

```suggestion
Previously in the  [Writing a CorDapp using a template](writing-a-cordapp-using-a-template) tutorial, you built a CorDapp which let you model IOUs on ledger.
```

**Review by erinking-r3 — COMMENTED** [review]
Anna, this is 💯   💯   💯 !!!  I left a few small suggestions.

File: content/en/docs/corda-os/4.8/modify-the-state.md [hunk 14]
```diff
@@ -11,36 +11,34 @@ menu:
...
-{{< tabs name="tabs-1" >}}
-{{% tab name="kotlin" %}}
-```kotlin
-interface ContractState {
-    // The list of entities considered to have a stake in this state.
-    val participants: List<AbstractParty>
-}
-```
-{{% /tab %}}
+* For your CorDapp, use the following `ContractState` interface:
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-state.md]: It's a step. Deleted the ul.

File: content/en/docs/corda-os/4.8/modify-the-state.md [hunk 15]
```diff
@@ -74,31 +71,28 @@ interface, your `IOUState` will also need properties to track the relevant featu
...
 
-If you’re following along in Java, you’ll need to first rename `TemplateState.java` to `IOUState.java`.
-
-To define `IOUState`, you'll also need to make the following changes:
-
-
-* You'll need to rename the `TemplateState` class to `IOUState`.
-* You'll need to add properties for `value`, `lender` and `borrower`, along with the required getters and setters in.
-Java:
+   a. Add properties for `value`, `lender` and `borrower`, along with the required getters and setters in Java:
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-state.md:83]: This is shown in the code block right below. I don't think I need to add any extra comments here.

File: content/en/docs/corda-os/4.8/run-your-cordapp.md [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+This will start a terminal window for each node. Give each node a moment to start - you’ll know it’s ready when its terminal windows displays
+the message “Welcome to the Corda interactive shell.”.
+
+
+{{< figure alt="running node" zoom="/en/images/running_node.png" >}}
+
+
+## Start the IOU flow
+
+Now that our nodes are running, order one of them to create an IOU by kicking off our `IOUFlow`. In a larger
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/run-your-cordapp.md]: Already updated with Ben's changes.

File: content/en/docs/corda-os/4.8/run-your-cordapp.md [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+   {{< /note >}}
+
+3. You want to create an IOU of 99 with `PartyB`. To start the `IOUFlow`, type the following syntax:
+
+   ```bash
+   start IOUFlow iouValue: 99, otherParty: "O=PartyB,L=New York,C=US"
+   ```
+
+   This single command will cause `PartyA` and `PartyB` to automatically agree an IOU. This is one of the great advantages of
+   the flow framework - it allows you to reduce complex negotiation and update processes into a single function call.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/run-your-cordapp.md:138]: No idea but I don't think it matters for the purposes of this tutorial.

File: content/en/docs/corda-os/4.8/tut-two-party-flow.md (hunk 8 above)
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/tut-two-party-flow.md]: Already applied Taylor's changes.

File: content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md [hunk 18]
```diff
@@ -11,44 +11,46 @@ menu:
...
-## Introduction
-
 CorDapps are applications that are installed on one or more Corda nodes, and that allow the node’s operator to instruct their node to perform some new process - anything from
 issuing a debt instrument to making a restaurant booking.
 
-Writing a CorDapp involves the following steps:
+Writing a CorDapp using a template involves the following steps:
+
+1. [Obtain the CorDapp template](obtain-the-cordapp-template)
+2. [Modify the state](modify-the-state.md)
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md:28]: This is a link, leaving as is.

File: content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md [hunk 19]
```diff
@@ -11,44 +11,46 @@ menu:
...
+3. [Modify the flow](modify-the-flow.md)
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md:29]: This is a link, leaving as is.

File: content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md [hunk 20]
```diff
@@ -11,44 +11,46 @@ menu:
...
+4. [Run your CorDapp](run-your-cordapp.md)
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md:30]: This is a link, leaving as is.

File: content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md [hunk 21]
```diff
@@ -11,44 +11,46 @@ menu:
...
-1. [Obtaining the CorDapp Template](hello-world-template.md)
-2. [Writing the state](hello-world-state.md)
-3. [Writing the flow](hello-world-flow.md)
-4. [Running your CorDapp](hello-world-running.md)
+## Before you start
 
-## Pre-requisites
+Before starting the tutorial steps, ensure you meet the following prerequisites:
+
+* [Familiarize yourself with Corda’s key concepts](key-concepts.md)
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md:37]: This is a link, leaving as is.

File: content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md [hunk 22]
```diff
@@ -11,44 +11,46 @@ menu:
...
+* [Get set up for CorDapp development](getting-set-up.md)
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md:38]: This is a link, leaving as is.

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 23]
```diff
@@ -11,13 +11,10 @@ menu:
...
 ---
 
-
-
-
-# Writing the flow
+# Modify the flow
 
 A flow encodes a sequence of steps that a node can perform to achieve a specific ledger update. Installing new flows
 on a node allows the node to handle new business processes. The flow that you define will allow a node to issue an
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:20]: applied the present tense

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 24]
```diff
@@ -61,11 +58,9 @@ to handle these tasks. Flows that are invoked in the context of a larger flow to
 *subflows*.
 
 
-## FlowLogic
+## Output
 
-All flows must subclass `FlowLogic`. You then define the steps taken by the flow by overriding `FlowLogic.call`.
-
-Let’s define your `IOUFlow`. Replace the definition of `Initiator` with the following:
+The following code presents how your template should look like after performing all the steps described below:
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:63]: Updated

[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:63]: Updated

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 25]
```diff
@@ -187,16 +197,31 @@ public class IOUFlow extends FlowLogic<Void> {
...
+
+3. Define your `IOUFlow`.
+
+{{< note >}}
+
+All flows must subclass `FlowLogic`. You then define the steps taken by the flow by overriding `FlowLogic.call`.
+
+{{< /note >}}
+
+Let’s walk through the above code step-by-step.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:218]: Updated

File: content/en/docs/corda-os/4.8/modify-the-state.md [hunk 26]
```diff
@@ -11,36 +11,34 @@ menu:
...
-interface ContractState {
-    // The list of entities considered to have a stake in this state.
-    val participants: List<AbstractParty>
-}
-```
-{{% /tab %}}
+* For your CorDapp, use the following `ContractState` interface:
 
-{{< /tabs >}}
+  {{< tabs name="tabs-1" >}}
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-state.md:29]: The user does nothing here. The ContractState interface is already there in the repo.

File: content/en/docs/corda-os/4.8/run-your-cordapp.md [hunk 27]
```diff
@@ -0,0 +1,239 @@
...
+|____logs
+|____network-parameters
+|____node.conf                // The node's configuration file.
+|____nodeInfo
+|____persistence.mv.db
+|____persistence.trace.db
+
+```
+
+1. In terminal, navigate to `build/nodes` from the root directory.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/run-your-cordapp.md:102]: Specified that this is the IntelliJ terminal window.

File: content/en/docs/corda-os/4.8/run-your-cordapp.md [hunk 28]
```diff
@@ -0,0 +1,239 @@
...
+1. Go to the terminal window displaying the CRaSH shell of `PartyA`.
+
+2. Type `help` to display a list of the available commands.
+
+   {{< note >}}
+   Local terminal shell is available only in a development mode. In a production environment SSH server can be enabled.
+   More about SSH and how to connect can be found on the [Node shell](shell.md) page.
+   {{< /note >}}
+
+3. You want to create an IOU of 99 with `PartyB`. To start the `IOUFlow`, type the following syntax:
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/run-your-cordapp.md:131]: Updated

File: content/en/docs/corda-os/4.8/run-your-cordapp.md [hunk 29]
```diff
@@ -0,0 +1,239 @@
...
+   transaction was not distributed to them.
+
+
+## Conclusion
+
+You have written a simple CorDapp that allows IOUs to be issued onto the ledger. This CorDapp is made up of two key
+parts:
+
+
+* The `IOUState`, which represents IOUs on the blockchain.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/run-your-cordapp.md:220]: Should be "on the ledger." Updated.

File: content/en/docs/corda-os/4.8/run-your-cordapp.md [hunk 30]
```diff
@@ -0,0 +1,239 @@
...
+## Next steps
+
+There are a number of improvements you could make to this CorDapp:
+
+* You could add unit tests, using the contract-test and flow-test frameworks.
+* You could change `IOUState.value` from an integer to a proper amount of a given currency.
+* You could add an API, to make it easier to interact with the CorDapp.
+
+But for now, the biggest priority is to add an `IOUContract` imposing constraints on the evolution of each
+`IOUState` over time - see [Applying contract constraints](tut-two-party-introduction.md).
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/run-your-cordapp.md:239]: Updated

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/tut-two-party-contract.md:64]: Specified which code block it is.

Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/tut-two-party-contract.md:64]: Updated

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md (hunk 11 above)
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/tut-two-party-contract.md:292]: ul not needed, updated the file

File: content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md [hunk 31]
```diff
@@ -11,44 +11,46 @@ menu:
...
 
-## Introduction
-
 CorDapps are applications that are installed on one or more Corda nodes, and that allow the node’s operator to instruct their node to perform some new process - anything from
 issuing a debt instrument to making a restaurant booking.
 
-Writing a CorDapp involves the following steps:
+Writing a CorDapp using a template involves the following steps:
+
+1. [Obtain the CorDapp template](obtain-the-cordapp-template)
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md:27]: Link, leaving as is.

File: themes/hugo-r3-theme/layouts/index.html [hunk 32]
```diff
@@ -602,25 +602,24 @@ <h2>CENM</h2>
...
         <li><a href=
-        "docs/corda-os/4.8/hello-world-introduction.html">Hello,
-        World!</a></li>
+        "docs/corda-os/4.8/writing-a-cordapp-using-a-template.html">Writing a CorDapp using a template pt 1
+</a></li>
 
         <li><a href=
-        "docs/corda-os/4.8/tut-two-party-introduction.html">Hello,
-        World! Pt 2 - Contract Constraints</a></li>
+        "docs/corda-os/4.8/tut-two-party-introduction.html">Writing a CorDapp using a template
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply themes/hugo-r3-theme/layouts/index.html:621]: Great idea. Updated the links.

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 33]
```diff
@@ -187,16 +197,31 @@ public class IOUFlow extends FlowLogic<Void> {
...
 
 You’ve defined your own `FlowLogic` subclass that overrides `FlowLogic.call`. `FlowLogic.call` has a return type
 that must match the type parameter passed to `FlowLogic` - this type is returned by running the flow.
 
 `FlowLogic` subclasses can optionally have constructor parameters, which can be used as arguments to
-`FlowLogic.call`. In our case, we have two:
-
+`FlowLogic.call`. In your case, you have two:
 
 * `iouValue`, which is the value of the IOU being issued.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:226]: updated

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 34]
```diff
@@ -187,16 +197,31 @@ public class IOUFlow extends FlowLogic<Void> {
...
 * `otherParty`, the IOU’s borrower (the node running the flow is the lender).
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:227]: added a real-life example to the use-case section

[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:227]: na

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 35]
```diff
@@ -216,24 +241,24 @@ Let’s walk through the steps of `FlowLogic.call` itself. This is where we actu
...
 
+Build your transaction proposal in two steps:
 
-* Creating the transaction’s components.
-* Adding these components to a transaction builder.
+1. Create the transaction’s components.
+2. Add these components to a transaction builder.
 
 
 #### Transaction items
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:264]: updated

File: content/en/docs/corda-os/4.8/modify-the-flow.md (hunk 9 above)
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:291]: updated

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 36]
```diff
@@ -261,7 +286,7 @@ Each `Command` contains a command type plus a list of public keys. For now, you
 the transaction to be valid, the lender is required to sign the transaction.
 
 
-#### Creating a transaction builder
+#### Create a transaction builder
 
 To actually build the proposed transaction, you need a `TransactionBuilder`. This is a mutable transaction class to
 which you can add inputs, outputs, commands, and any other items the transaction needs. You create a
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:292]: updated

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 37]
```diff
@@ -277,7 +302,7 @@ passing in a reference to the `TemplateContract`, which imposes no constraints.
 real constraints in the next tutorial.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:302]: updated

File: content/en/docs/corda-os/4.8/modify-the-flow.md (hunk 10 above)
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:317]: updated

File: content/en/docs/corda-os/4.8/modify-the-flow.md (hunk 6 above)
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md:382]: Added the link.

File: content/en/docs/corda-os/4.8/modify-the-state.md [hunk 38]
```diff
@@ -61,11 +59,10 @@ If you do want to dive into Kotlin, see the official
...
-## Modelling IOUs
+## Define IOUState
 
 How should you define the `IOUState` representing IOUs on the blockchain? Beyond implementing the `ContractState`
-interface, your `IOUState` will also need properties to track the relevant features of the IOU:
-
+interface, your `IOUState` also needs properties to track the relevant features of the IOU:
 
 * The value of the IOU.
 * The lender of the IOU.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-state.md:68]: here the lender and the borrower are listed as the IOU features. Leaving as is.

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md [hunk 39]
```diff
@@ -213,7 +231,7 @@ to one redeeming an IOU.
 * They allow you to define the required signers for the transaction. For example, IOU creation might require signatures
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/tut-two-party-contract.md:231]: This is an ul item. Leaving as is.

File: content/en/docs/corda-os/4.8/tut-two-party-flow.md (hunk 7 above)
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/tut-two-party-flow.md:23]: updated

File: content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md [hunk 40]
```diff
@@ -11,44 +11,46 @@ menu:
...
+Before starting the tutorial steps, ensure you meet the following prerequisites:
+
+* [Familiarize yourself with Corda’s key concepts](key-concepts.md)
+* [Get set up for CorDapp development](getting-set-up.md)
+* [Run your first CorDapp](tutorial-cordapp.md)
 
-Before you begin, [your dev environment should be set up](getting-set-up.md), you should have run
-[your first CorDapp](tutorial-cordapp.md), and should be familiar with Corda’s [key concepts](key-concepts.md).
 
 ## Use-case
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md:42]: yeah, looks weird. will change to use case.

File: content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md [hunk 41]
```diff
@@ -11,44 +11,46 @@ menu:
...
 
 
-# Hello, World!
+# Writing a CorDapp using a template
 
 Ready to write your first CorDapp? You've come to the right place!
 
-## Introduction
-
 CorDapps are applications that are installed on one or more Corda nodes, and that allow the node’s operator to instruct their node to perform some new process - anything from
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/writing-a-cordapp-using-a-template.md:22]: Reused Erin's CorDapp description here.

File: content/en/docs/corda-os/4.8/tut-two-party-introduction.md (hunk 12 above)
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/tut-two-party-introduction.md]: It should be "on ledger".

File: content/en/docs/corda-os/4.8/modify-the-flow.md [hunk 42]
```diff
@@ -187,16 +197,31 @@ public class IOUFlow extends FlowLogic<Void> {
...
 
-If you’re following along in Java, you’ll also need to rename `Initiator.java` to `IOUFlow.java`.
 
-Let’s walk through this code step-by-step.
+## Define FlowLogic
+
+1. Depending whether you are working with Java or Kotlin template, perform one of the following steps:
+
+   * For Java, open `Initiator.java` from `workflows/src/main/java/com/template/flows/Initiator.java`.
+   * For Kotlin, open `Flows.kt` from `workflows/src/main/kotlin/com/template/flows/Flows.kt`.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/modify-the-flow.md]: updated

File: content/en/docs/corda-os/4.8/tut-two-party-contract.md (hunk 3 above)
Comment (AnnaK-R3) [reply content/en/docs/corda-os/4.8/tut-two-party-contract.md]: updated

**Review by erinking-r3 — APPROVED** [review]
LGTM!

**Pull request merged**